virttest/env_process.py

```diff
@@ -1096,18 +1097,13 @@ def preprocess(test, params, env):
     _setup_manager.register(StorageConfig)
     _setup_manager.register(FirewalldService)
     _setup_manager.register(IPSniffer)
+    _setup_manager.register(RegisterMigrateVMs)
     _setup_manager.do_setup()
 
     vm_type = params.get("vm_type")
 
     base_dir = data_dir.get_data_dir()
 
-    # Add migrate_vms to vms
-    migrate_vms = params.objects("migrate_vms")
-    if migrate_vms:
-        vms = list(set(params.objects("vms") + migrate_vms))
-        params["vms"] = " ".join(vms)
```

Comment:
Let's skip this part since it should be part of the VM process: https://github.com/avocado-framework/avocado-vt/blob/b411ac28f5abda64de16cea97ff906e05013b8f0/virttest/env_process.py#L883-L886
Actually,  this part does not relate to the environment of the host setupper, just add the parameters `migrate_vms` to the global `vms`.
As for the previous setuppers, we can see that mainly set up the host environments, such as Networking, Bridge, and so on.
So I prefer to skip this part. What about your opinions? @luckyh 
Please let me know your thoughts @bgartzi Thanks.

Reply:
Hi @YongxueHong!

Yeah, I get your point. I was considering these kind of pieces of code as part of the refactoring. The reason for that was that changing the order they are ran could break the order in which each of these steps are executed. `Setuper`'s `self.params` is just a reference, so updating it's keys and values affect directly to the dictionary being used inside the `preprocess` or `postprocess` function. 

Should we run this piece of code in the beginning of the `preprocess` function instead?
Should we treat these kind of steps (tweaking test parameter config) in a different manner?

Thanks for the feedback and let me know your thoughts. Let's also see what @luckyh thinks about it.

Reply:
> Should we run this piece of code in the beginning of the preprocess function instead?

@bgartzi @YongxueHong it looks that the piece of code is relocatable so I'd like to prefer this option.

> Should we treat these kind of steps (tweaking test parameter config) in a different manner?

That's a good point. From my perspective, we did require a mechanism for resolving params (e.g. merge, concatenate, etc.) before running into the `preprocess` function. However, this would also require some new abstracts to be introduced, therefore I'd suggest we could skip it for this time.

What do you think?

Reply:
I agree. I'm relocating it to... right after the `preprocess_vm_off_hook` call. Setup steps between that and the `_setup_manager` initialization don't seem to rely on `params['vms']`, so it looks safe.

I'm pushing the patch now.